src/utils/functions/fetchData.js

```diff
@@ -0,0 +1,8 @@
+const fetchData = async (API_URL) => {
+  const response = await fetch(API_URL);
+  const data = await response.json();
```

Comment:
When using `await` always wrap the code in `try catch` block. If the `awaited` function throws an exception, the rest of the execution will be truncated, causing some weird behaviors in your application. Also, consider that `fetch` won't throw an error if the request is a HTTP error code, not even on 404. The only way `fetch` will throw is due to connection issues, or not able to perform the request at all. Check:

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#differences_from_jquery